libs/core/async_cuda/include/hpx/async_cuda/detail/cuda_event_callback.hpp

```diff
@@ -26,7 +26,7 @@ namespace hpx { namespace cuda { namespace experimental { namespace detail {
         hpx::move_only_function<void(cudaError_t)>;
 
     HPX_CORE_EXPORT void add_event_callback(
-        event_callback_function_type&& f, cudaStream_t stream);
+        event_callback_function_type&& f, cudaStream_t stream, int device);
```

Comment:
Let's keep this API function backwards compatible as well:
```suggestion
        event_callback_function_type&& f, cudaStream_t stream, int device = 0);
```